open-craft/learner-profile-dashboard
# Address post-delivery feedback

cf. [OC-4550](https://tasks.opencraft.com/browse/OC-4550)

This PR addresses all feedback that we got from the TLL team so far, as listed on the ticket. (It might make sense to step through the diff by checking individual commits. The PR includes one commit per feedback item, plus a couple commits correcting typos and copy-paste errors.) Exceptions are items 2, 3, and 4: We had already covered the second item via #13, and the third and fourth item just change the content of two different questions for the 2018 version of the HPL course. Since this repository does not include data for the 2018 version of the HPL course I applied the content changes directly to LPD stage and prod.

**Test instructions**

1. Run migrations.

1. Collect static assets.

1. Step through the feedback items on the ticket and verify that they have been addressed.

    - You can skip items 2, 3, and 4 at this stage.

    - As a general note, only answer questions that are not configured to `influence_recommendations`/`influence_group_membership` for the purpose of testing the different features. Otherwise, the fact that your local installation is not set up to connect to an adaptive engine deployment might interfere (i.e., you might not get the expected results because the transmission of learner data to the adaptive engine will fail).

    - For item 6, submit an answer to a qualitative question that is of type `"short-answer"` and check the DB for the corresponding `QualitativeAnswer`. (There should only be one answer.) Then set `split_answer` to `True` for the qualitative question you just answered, change your answer (making sure that it includes one or more commas), and submit again. The number of `QualitativeAnswer` objects for the question should be equal to `n+1` where `n` represents the number of commas in your answer. The `QualitativeAnswer` that was created prior to setting `split_answer` to `True` should no longer exist in the DB. *In general*, the LPD will delete any answer content that does not match a learner's most recent answer to a qualitative question. This means that, for a given qualitative question and learner, the DB should only ever contain:

        - zero `QualitativeAnswer` objects (if the learner has never answered the question), or

        - one `QualitativeAnswer` object (if the learner answered the question at least once and it has `split_answer` set to `False`), or

        - `n+1` `QualitativeAnswer` objects, where `n` represents the number of commas in the learner's answer (if the learner answered the question at least once and it has `split_answer` set to `True`).

    - For item 7, make sure that text next to "Submit your answers" button says "Not yet submitted" before you change some answers and click the button for the first time.

        - Once you click the button, the text should update to mention the current time and date.

        - After refreshing the page, the text should mention the time and date of your most recent submission (which may or may not match the current time and date at this point 😄)

1. Finally, to verify items 2 and 3, double-check question texts of updated questions on LPD stage and prod.

**Reviewers**
